spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DnnTensor.scala

```diff
@@ -95,6 +95,11 @@ class DnnTensor[T: ClassTag](
     Memory.Axpby(this.nElement(), a, x, b, y)
   }
 
+  def scale(from: DnnTensor[T], scal: Float): Unit = {
+    val length = this.nElement()
+    Memory.Scale(length, scal, from._storage.ptr.address, this._storage.ptr.address)
```

Comment:
Should we create a new Class to handle all math operations instead of adding them to Memory?

Reply:
Yes. We can do it. But we should refactor `BigDL-core` first.